docs/reference/mapping/types/range.asciidoc

```diff
@@ -0,0 +1,107 @@
+[[range]]
+=== Range datatypes
+
+The following range types are supported:
+
+[horizontal]
+`integer_range`::   A range of signed 32-bit integers with a minimum value of +-2^31^+ and maximum of +2^31^-1+.
+`float_range`::     A range of single-precision 32-bit IEEE 754 floating point values.
+`long_range`::      A range of signed 64-bit integers with a minimum value of +-2^63^+ and maximum of +2^63^-1+.
+`double_range`::    A range of double-precision 64-bit IEEE 754 floating point values.
+`date_range`::      A range of date values represented as unsigned 64-bit integer milliseconds elapsed since system epoch.
+
+Below is an example of configuring a mapping with various range fields followed by an index and range query example:
+
+[source,js]
+--------------------------------------------------
+PUT my_index
+{
+  "mappings": {
+    "my_type": {
+      "properties": {
+        "expected_attendees": {
+          "type": "integer_range"
+        },
+        "time_frame": {
+          "type": "date_range", <1>
+          "format": "yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis"
+        },
+        "price_range": {
+          "type": "float_range"
+        }
+      }
+    }
+  }
+}
+
+PUT my_index/my_type/1
+{
+  "expected_attendees" : { <2>
+    "gte" : 10,
+    "lte" : 20
+  },
+  "time_frame" : { <3>
+    "gte" : "2015-10-31 12:00:00", <4>
+    "lte" : "2015-11-01"
+  },
+  "price_range" : { <5>
+    "gt" : 59.99,
+    "lt" : 100.00
+  }
+}
+
+GET my_index/_search
+{
+  "range" : {
+    "time_frame" : { <6>
+      "gte" : "2015-10-31",
+      "lte" : "2015-11-01 12:00:00",
+      "relation" : "within" <7>
+    }
+  }
+}
+--------------------------------------------------
+// TESTRESPONSE
```

Comment:
You should have a `// CONSOLE` here, and the `// TESTRESPONSE` block should go under another source code block which contains the results of the query.  This block can be inside a comment block as below so that it doesn't render in the docs, eg:

    //////////

    [source,js]
    ----------
    { 
      "took": 25,
      "hits": ...
    }
    ----------
    // TESTRESPONSE[s/"took": 25/"took": $body.took/]

    //////////